lib/omnibus/packagers/ociru.rb

```diff
@@ -0,0 +1,282 @@
+#
+# Copyright 2014 Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "find"
+require "json"
+require "pathname"
+require "omnibus/packagers/windows_base"
+require "fileutils"
+
+module Omnibus
+  class Packager::OCIRU < Packager::Base
+    id :ociru
+    intermediate_pkg_name = "package.tar.xz"
+
+    setup do
+    end
+
+    build do
+      # create the payload directory, copy the install_dir and extra files to it
+      # TODO: does the makedirs do what we need in terms of file permissions?
+      payload_dir = File.join(staging_dir, "payload")
+      install_dir = File.join(payload_dir, project.install_dir)
+      FileSyncer.sync(project.install_dir, install_dir, exclude: exclusions)
+      project.extra_package_files.each do |file|
+        if File.directory?(file)
+          destination = File.join(payload_dir, file)
+          FileUtils.makedirs(destination)
+          FileSyncer.sync(file, destination)
+        else
+          destination = File.join(payload_dir, File.dirname(file))
+          FileUtils.makedirs(destination)
+          FileUtils.cp(file, destination, preserve: true)
+        end
+      end
+      fl = filelist(payload_dir)
+
+      compress_env = { "XZ_OPT" => "-T#{compression_threads} -#{compression_level}" }
+
+      # create the archive
+      archive_file = windows_safe_path(staging_dir, intermediate_pkg_name)
```

Comment:
pardon my lack of knowledge, is `windows_safe_path` to be used whatever the OS?

Reply:
yeah, it basically joins path using the correct slash based on the platform